### PR TITLE
Add metrics correctness support to makefile + ci

### PR DESCRIPTION
**Pull request by pmcollins**
This change adds two makefile targets, a CI job to run the metrics correctness test, and functionality to generate an .md file with the results of the test.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -170,6 +176,7 @@ workflows:
           requires:
             - cross-compile
             - correctness
```
Comment (bogdandrutu) [inline .circleci/config.yml:178]: Should this be renamed to `correctness_trace`?

Comment (pmcollins) [reply .circleci/config.yml:178]: Yeah, I think so, but was thinking in a separate PR.

Comment (bogdandrutu) [reply .circleci/config.yml:178]: Ok

**Comment by bogdandrutu** [comment]
@pmcollins are you going to add this to github actions maybe? we want to deprecate the circle CI, and already have actions for correctness.

**Comment by tigrannajaryan** [comment]
@bogdandrutu @pmcollins should we merge this as is while CircleCI is still there? We will need a follow up PR to add this toe to Github actions.

**Comment by tigrannajaryan** [comment]
> @bogdandrutu @pmcollins should we merge this as is while CircleCI is still there? We will need a follow up PR to add this toe to Github actions.

@bogdandrutu @pmcollins gentle ping. :-)

**Comment by pmcollins** [comment]
Either way is fine with me :-)

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@pmcollins please do a followup to move to github actions

**Comment by bogdandrutu** [comment]
Also you need to rebase so I can merge this (a required check was added later and is not running)

**Pull request merged**